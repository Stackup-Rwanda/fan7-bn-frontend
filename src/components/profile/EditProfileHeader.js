import React, { Component } from 'react';
import './styles/editProfileHeader.scss';

class EditProfileHeader extends Component {
  constructor(props) {
    super(props);
    this.initialState = {
      isActive: 'profile',
    };
    this.state = this.initialState;
    this.toggleActiveClass = this.toggleActiveClass.bind(this);
  }

  toggleActiveClass(event) {
    this.setState({
      isActive: event.target.name,
    });
  }

  render() {
    const { isActive } = this.state;

    return (
      <div className="header">
        <h1 className="header__title">Settings</h1>
        <div className="header__menu">
          <button
            type="button"
            className={
              isActive === 'profile' ? 'header__menu___item active' : 'header__menu___item'
            }
            name="profile"
            onClick={this.toggleActiveClass}
          >
            Profile
          </button>
          <button
            type="button"
            className={
              isActive === 'notification' ? 'header__menu___item active' : 'header__menu___item'
            }
            name="notification"
            onClick={this.toggleActiveClass}
          >
            Notification
          </button>
        </div>
      </div>
    );
  }
}

export default EditProfileHeader;
