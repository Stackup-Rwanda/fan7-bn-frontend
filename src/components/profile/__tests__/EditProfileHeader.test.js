import React from 'react';
import { mount, shallow } from 'enzyme';
import EditProfileHeader from '../EditProfileHeader';

describe('EditProfileHeader Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(<EditProfileHeader />);

    expect(wrapper.find('button[name="profile"]').exists()).toBe(true);
    expect(wrapper.find('button[name="notification"]').exists()).toBe(true);
    expect(wrapper.state().isActive).toEqual('profile');
  });

  it('should handle toggle active menu', () => {
    const event = {
      target: {
        name: 'notification',
      },
    };
    const wrapper = shallow(<EditProfileHeader />);

    wrapper.find('button[name="notification"]').simulate('click', event);

    expect(wrapper.state().isActive).toEqual('notification');
  });
});

describe('<EditProfileHeader /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<EditProfileHeader />);
    expect(tree).toMatchSnapshot();
  });
});
