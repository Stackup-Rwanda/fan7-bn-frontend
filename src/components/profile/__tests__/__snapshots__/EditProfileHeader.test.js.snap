// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfileHeader /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  >
    Settings
  </h1>
  <div
    className="header__menu"
  >
    <button
      className="header__menu___item active"
      name="profile"
      onClick={[Function]}
      type="button"
    >
      Profile
    </button>
    <button
      className="header__menu___item"
      name="notification"
      onClick={[Function]}
      type="button"
    >
      Notification
    </button>
  </div>
</div>
`;
